arwen/host/execution.go

```diff
@@ -251,18 +260,25 @@ func (host *vmHost) ExecuteOnDestContext(input *vmcommon.ContractCallInput) (vmO
 
 func (host *vmHost) handleBuiltinFunctionCall(input *vmcommon.ContractCallInput) (*vmcommon.ContractCallInput, *vmcommon.VMOutput, error) {
 	output := host.Output()
+
+	asyncPrefixArgs := arwen.PopCallIDsFromArguments(input)
+
 	postBuiltinInput, builtinOutput, err := host.callBuiltinFunction(input)
 	if err != nil {
 		log.Trace("ExecuteOnDestContext builtin function", "error", err)
 		return nil, nil, err
 	}
 
+	if postBuiltinInput != nil {
+		arwen.PrependToArguments(postBuiltinInput, asyncPrefixArgs...)
+	}
+
 	output.AddToActiveState(builtinOutput)
 
 	return postBuiltinInput, builtinOutput, nil
 }
 
-func (host *vmHost) executeOnDestContextNoBuiltinFunction(input *vmcommon.ContractCallInput) (vmOutput *vmcommon.VMOutput, err error) {
+func (host *vmHost) executeOnDestContextNoBuiltinFunction(input *vmcommon.ContractCallInput) (vmOutput *vmcommon.VMOutput, isChildComplete bool, err error) {
```

Comment:
Named returns are not referenced in the body of the function; can be deleted.

Reply:
vmOutput is referenced